source/Grid/Grid.js

```diff
@@ -496,6 +496,47 @@ export default class Grid extends Component {
     )
   }
 
+  renderColumns (childrenToDisplay, rowIndex, top) {
+    let { columnsLocked } = this.props
```

Comment:
I'm not sure if there was a strong reason to inline all the rendering, but the repetition now was getting out of hand so I moved them to separate functions